cmd/fleetctl/mdm.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/service"
+	kithttp "github.com/go-kit/kit/transport/http"
+	"github.com/urfave/cli/v2"
+)
+
+func mdmCommand() *cli.Command {
+	return &cli.Command{
+		Name:  "mdm",
+		Usage: "Run MDM commands against your hosts",
+		Flags: []cli.Flag{
+			configFlag(),
+			contextFlag(),
+			debugFlag(),
+		},
+		Subcommands: []*cli.Command{
+			mdmRunCommand(),
+		},
+	}
+}
+
+func mdmRunCommand() *cli.Command {
+	return &cli.Command{
+		Name:    "run-command",
+		Aliases: []string{"run_command"},
+		Usage:   "Run a custom MDM command on one macOS host. Head to Apple's documentation for a list of available commands and example payloads here:  https://developer.apple.com/documentation/devicemanagement/commands_and_queries",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:     "host",
+				Usage:    "The host, specified by hostname, uuid, osquery_host_id or node_key, that you want to run the MDM command on.",
+				Required: true,
+			},
+			&cli.StringFlag{
+				Name:     "payload",
+				Usage:    "A path to an XML file containing the raw MDM request payload.",
+				Required: true,
+			},
+		},
+		Action: func(c *cli.Context) error {
+			client, err := clientFromCLI(c)
+			if err != nil {
+				return fmt.Errorf("create client: %w", err)
+			}
+
+			// print an error if MDM is not configured
+			if err := checkMDMEnabled(client); err != nil {
+				return err
+			}
+
+			hostIdent := c.String("host")
+			payloadFile := c.String("payload")
+			payload, err := os.ReadFile(payloadFile)
+			if err != nil {
+				return fmt.Errorf("read payload: %w", err)
+			}
+
+			host, err := client.HostByIdentifier(hostIdent)
+			if err != nil {
+				var nfe service.NotFoundErr
+				if errors.As(err, &nfe) {
+					return errors.New("The host doesn't exist. Please provide a valid hostname, uuid, osquery_host_id or node_key.")
+				}
+				var sce kithttp.StatusCoder
+				if errors.As(err, &sce) {
+					if sce.StatusCode() == http.StatusForbidden {
+						return fmt.Errorf("Permission denied. You don't have permission to run an MDM command on this host: %w", err)
+					}
+				}
+				return err
+			}
+
+			// TODO(mna): this "On" check is brittle, but looks like it's the only
+			// enrollment indication we have right now...
+			if host.MDM.EnrollmentStatus == nil || !strings.HasPrefix(*host.MDM.EnrollmentStatus, "On") ||
+				host.MDM.Name != fleet.WellKnownMDMFleet {
```

Comment:
agreed, sounds like in the future this check should come from the API directly, we're also using similar logic in the UI (`enrollment_status == on && mdm.name == "fleet"`)